src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -2021,6 +2021,31 @@ class RGWBucketMetadataHandler : public RGWBucketMetadataHandlerBase {
     if (ret < 0)
       return ret;
 
+    map<string, bufferlist> attrs_m;
+    RGWBucketInfo bi;
+
+    ret = ctl.bucket->read_bucket_instance_info(be.bucket, &bi, y, dpp, RGWBucketCtl::BucketInstance::GetParams()
+                                                                    .set_mtime(&orig_mtime)
+                                                                    .set_attrs(&attrs_m));
+    if (ret < 0) {
+        return ret;
+    }
+
+    if (bi.deleted) {
+      bi.layout.logs.push_back(rgw::log_layout_from_deleted_index(0, bi.layout.current_index));
+    }
```

Comment:
I can't seem to exercise the call path to do_remove() of RGWBucketInstanceMetadataHandler. I think I am missing something.

Reply:
it seems like we don't call into `RGWBucketInstanceMetadataHandler::do_remove()` because the metadata_key we pass in through `RGWMetadataManager::remove() ` is  without an instance like metadata_key="bucket:some-bucket". whereas in metadata put() call we pass in the bucket instance.

Reply:
right, bucket deletion in multisite only deletes the bucket entrypoint metadata, so that's the "bucket:some-bucket" key you're seeing

for the bucket instance metadata, the primary just adds the `BUCKET_DELETED` flag, so metadata sync would see that in `RGWBucketInstanceMetadataHandler::do_put()` -> `RGWMetadataHandlerPut_BucketInstance` instead